core/src/main/java/software/amazon/awssdk/interceptor/ExecutionInterceptor.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.interceptor;
```

Comment:
Request Handler = Execution Interceptor

Request -> Execution because it's also used for responses and works within the scope of a single execution (a concept they'll need to understand if they're going to implement one).

Handler -> Interceptor because what the heck is a handler? Isn't pretty much everything a handler? Interceptor seems to be the predominant word used for this type of pattern.